docs: adopt latest Charmcraft profiles, incl juju_setup marker on integration tests (#2497)
canonical/operator: docs/howto/write-integration-tests-for-a-charm.md

Tests run sequentially in the order they are written in the file. It can be useful to put tests that deploy applications in the top of the file so the applications can be used by subsequent tests.

> We recommend that you mark such tests with `@pytest.mark.juju_setup`. If you later use `--no-juju-setup` to skip the deployment tests, the model must already exist. See {ref}`write-integration-tests-for-a-charm-run-your-tests`.

Reviewer: How about:
```suggestion
Mark these tests with `@pytest.mark.juju_setup`. Then you can use `--no-juju-setup` with `--juju-model` to skip the deployment tests and run against an existing model. See {ref}`write-integration-tests-for-a-charm-run-your-tests`.
```
